src/gopher.cc

```diff
@@ -598,9 +604,13 @@ gopherToHTML(GopherStateData * gopherState, char *inbuf, int len)
 
                     } else if (gtype == GOPHER_INFO) {
                         snprintf(tmpbuf, TEMP_BUF_SIZE, "\t%s\n", html_quote(name));
+                    } else if (gtype == GOPHER_WWW) {
+                        snprintf(tmpbuf, TEMP_BUF_SIZE, "<IMG border=\"0\" SRC=\"%s\"> <A HREF=\"%s\">%s</A>\n",
+                                     icon_url, rfc1738_escape_part(selector), html_quote(name));
```

Comment:
Hm. Should not we be adding the URI scheme and `host` to the `HREF` value? Or is `w` used for _relative_ paths on the same Gopher server (implying a `gopher://` scheme, the current host authority, and the current base path).

Is there a good web page that documents the expected selector (and other details) for the `w` type? My quick search did not yield any usable results for that specific type.

Reply:
AFAICT, your test example at `gopher://gopher.zcrayfish.soy/1/irc/test` appears to imply that the selector contains a regular URL with scheme, authority, and path:

```
curl gopher://gopher.zcrayfish.soy/1/irc/test
...
wCNN HTTPS      http://www.cnn.com/     (NULL)  1
wzdo gopher     gopher://gopher.zcrayfish.soy/  (NULL)  1
```

If these `w` selectors are full URLs, then we cannot encode them with rfc1738_escape_part() because that would encode essential delimiters like `:` and `/` (they are a part of the `RFC1738_ESCAPE_RESERVED` set), right?

Reply:
> Is there a good web page that documents the expected selector (and other details) for the `w` type? My quick search did not yield any usable results for that specific type.

Not that I know of... I'm just going by the libwww source (which supported the `w` type as far back as 1992) and every real world example I've ever seen; in which the selector has been a full URL.


> If these `w` selectors are full URLs, then we cannot encode them with rfc1738_escape_part() because that would encode essential delimiters like `:` and `/` (they are a part of the `RFC1738_ESCAPE_RESERVED` set), right?

I think rfc1738_escape_unescaped() would work a lot better there, your thoughts?

Reply:
Each part of the URL has different encoding requirements. This is why I think using AnyP::Uri will be safer - detection of non-URLs, incomplete URLs, and RFC 3986 encoding of each part of the URL.

Reply:
> > Alex: Is there a good web page that documents the expected selector (and other details) for the `w` type? My quick search did not yield any usable results for that specific type.

> Zachary: Not that I know of... I'm just going by the libwww source (which supported the `w` type as far back as 1992) and every real world example I've ever seen; in which the selector has been a full URL.

I added that critical information to the PR description.
 
> I think rfc1738_escape_unescaped() would work a lot better there, your thoughts?

I agree. Its documentation matches this code needs AFAICT: "Suitable for safely encoding an absolute URL which may be encoded but is not trusted."